pkg/mqtt/client.go

```diff
@@ -0,0 +1,135 @@
+package mqtt
+
+import (
+	"fmt"
+	"path"
+
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog/log"
+)
+
+const (
+	Online  string = "online"
+	Offline string = "offline"
+)
+
+// Topics.
+const (
+	State        string = "state"
+	Command      string = "command"
+	Event        string = "event"
+	serverStatus string = "server/status"
+)
+
+type Client interface {
+	// Connect to the MQTT server.
+	Connect() error
+	// Disconnect from the MQTT server.
+	Disconnect() error
+
+	// Publishes a message under the prefix topic of DigitalStrom.
+	Publish(topic string, message interface{}) error
+	// Subscribe to a topic and calls the given handler when a message is
+	// received.
+	Subscribe(topic string, messageHandler mqtt.MessageHandler) error
+
+	// Return the full topic for a given subpath.
+	GetFullTopic(topic string) string
+	// Returns the topic used to publish the server status.
+	ServerStatusTopic() string
+
+	RawClient() mqtt.Client
```

Comment:
The exposure of the raw client is because this client abstracts all the share logic for the topics related to DigitalStrom (e.g. topic prefix) but to publish the Home Assistant discovery messages you need to have full control from the raw client.